rslint_core/src/groups/errors/valid_typeof.rs

```diff
@@ -0,0 +1,116 @@
+use crate::rule_prelude::*;
+use ast::{BinExpr, Expr, UnaryOp};
+use rslint_parser::{SyntaxText, TextRange};
+use SyntaxKind::*;
+
+declare_lint! {
+    /**
+    Enforces the to use valid string literals in a `typeof` comparison.
+
+    ## Invalid Code Examples
+    ```ignore
+    typeof foo === "strnig"
+    typeof foo == "undefimed"
+    typeof bar != "nunber"
+    typeof bar !== "fucntion"
+    ```
+    */
+    #[derive(Default)]
+    ValidTypeof,
+    errors,
+    "valid-typeof"
+}
+
+const VALID_TYPES: [&str; 8] = [
+    "undefined",
+    "object",
+    "boolean",
+    "number",
+    "string",
+    "function",
+    "symbol",
+    "bigint",
+];
+
+#[typetag::serde]
+impl CstRule for ValidTypeof {
+    fn check_node(&self, node: &SyntaxNode, ctx: &mut RuleCtx) -> Option<()> {
+        if node.kind() == BIN_EXPR {
+            let expr = node.to::<BinExpr>();
```

Comment:
```suggestion
        let expr = node.try_to::<BinExpr>()?;
```
Since its checking one thing you can simply shorten it to this